ts/screens/authentication/cie/CieCardReaderScreen.tsx

```diff
@@ -124,16 +149,27 @@ class CieCardReaderScreen extends React.PureComponent<Props, State> {
     return this.props.navigation.getParam("authorizationUri");
   }
 
-  private setError = (
-    errorMessage: string,
-    navigationRoute?: string,
-    navigationParams: Record<string, unknown> = {}
-  ) => {
-    this.dispatchAnalyticEvent(errorMessage);
+  private setError = ({
+    eventReason,
+    errorDescription,
+    navigationRoute,
+    navigationParams = {}
+  }: setErrorParameter) => {
+    const cieDescription = errorDescription
```

Comment:
```suggestion
    const cieDescription = errorDescription ??
      fromNullable(analyticActions.get(eventReason)).getOrElse("")
```
A double ternary operator could be difficult to read.
Could it be an improvement?